### ISSUE TITLE
Purpose of this fork - Plans for future maintenance/development

### ISSUE THREAD

**Issue by akien-mga** [LibreSprite/LibreSprite#1]
Since https://github.com/aseprite/aseprite went proprietary on Aug 26th, 2016 (https://github.com/aseprite/aseprite/commit/5ecc356a41c8e29977f8608d8826489d24f5fa6c), this fork and corresponding GitHub organization were created to preserve the last GPLv2 version of the code.

Our reasons for doing so are multiple and likely different for all fork initiators, but a strong belief in free software is definitely to name, and as the GPLv2 license allows us to, we want to safeguard this nice open source piece of software.

---
### Development intent

We now have to discuss our plans for the future. There are several possibilities:
- Continue the development in parallel to the proprietary upstream version (i.e. effectively become a competitive upstream)
- Do minimal maintenance of the project, merging good PRs and doing some simple bug fixes here and there but no strong development
- Keep the repository as a more or less read-only archive of the code as it was before going proprietary, so that other interested parties can build it for themselves, for distribution, or fork it to continue development as a new upstream
### Providing binaries

In parallel to this, we have the possibility to set up a small website where we would provide binaries for Linux, Mac OSX and Windows of the latest stable release and likely of the current master HEAD. These would be useful for people who prefer to use the libre version over the proprietary one but can't/don't want to build from source.

We can also choose not to do so, this is up for discussion. Of course, the GPLv2 allows us to distribute such binaries.
### Branding

Depending on the decisions for the above points, we might end up being disliked by the original author and developer of the now proprietary version. I see no trademark policy so far, but it might come to a point where we need to rebrand this fork to avoid confusion and unnecessary friction.

If a new name is needed, it should be discussed in a new issue, please don't post alternative name suggestions on this topic.

Regardless of a potential name change, if we decide to further develop or at least maintain this fork, the various links and statements in the documentation files will have to be changed, to avoid having e.g. people reporting our own bugs to the proprietary upstream repo.

I already removed the EULA of the original developer, as it applied to his repository and not to the GPLv2 code itself. We obviously don't want to keep its terms which are unnecessary for a GPLv2 project.

**Comment by akien-mga** [comment]
Please express your opinions on the above matter, and especially tell us if you'd like to join the effort. The purpose of this repository and GitHub organisation is to concentrate the effort of all developers who might be interested in seeing a GPLv2 community fork to the now proprietary aseprite.

**Comment by eska014** [comment]
I've been using Aseprite before the license change, and I'm interested in doing some changes. Haven't worked with the code before though, due to the original repository's CLA and reluctance to maintain a fork by myself

I developed a Godot Engine import plug-in for Aseprite animations (https://github.com/eska014/aseprite-import/), and Aseprite's export tools could use some changes to improve the workflow (not only for this plug-in).

I'll try and fix reported bugs as well I can, but Aseprite isn't really my main tool of trade, so I can only invest so much time

By the way, this is the first comment from the developer on the license change I found: https://twitter.com/davidcapello/status/771051291896377344
I'd generally support distributing binaries, but I don't understand what the issue is here

**Comment by zachwick** [comment]
I've been a semi-casual user of Aseprite for some time, and I was also very saddened by this licensing change.

I am willing to commit some development time to both maintenance (reviewing and merging PRs, simple bug fixes, etc.) and continuing forward development as a competing upstream.

As for providing binaries, I am of the opinion that it is of immense value to provide binaries of this libre version and would be willing to help do that also. In that vein, I am bullish on providing binaries of both a stable version and a "nightly" version. I'm unsure if a standalone website is required for distributing these binaries as Github does have a "Releases" component in which a repo can provide direct download links to archives.

I do have some thoughts on branding. I am not a lawyer, but my guess is that we probably cannot call this fork aseprite-gpl in the long term. I agree that this name discussion should happen in a different issue. The name change would also obviously inform the URL of any site that gets set up to direct users to for this libre fork.

**Comment by akien-mga** [comment]
As for myself, my main motivation with this fork is to gather all potentially interested contributors in one place. I package aseprite for the Mageia Linux distro, and otherwise don't use it much (yet - I'd like to play with it a bit but time is sparse). As such, I don't particularly intend to do advanced development work myself, but I can help with the maintenance effort, reviewing some PRs, triaging some issues.

I would give this discussion a few days to let all interested parties express their opinion, and also to see how the original author reacts to the various discussions running now (especially on his Twitter) regarding this fork. There seems to be a strong misunderstanding of what the open source and Linux ecosystems are, which led to frustration and the change of license. It could be that the author changes his mind again, which is why I'd recommend a "wait and see" approach for the next few days.

That said, the original upstream CLA really crippled the project even as GPLv2, so a competing fork without CLA and with a guarantee to stay GPLv2 forever might be a sensible path to go on.

**Comment by FaTony** [comment]
Never used Aserprite but I think keeping GPL version is a very good cause.

**Comment by akien-mga** [comment]
Here are the original developer's thoughts on the license change: http://dev.aseprite.org/post/149797781837/new-source-code-license

He makes a few good points IMO, but I hardly see that as justifying removing the GPLv2 license, which wasn't harming the project in any way.

**Comment by JeromeMartinez** [comment]
> but I hardly see that as justifying removing the GPLv2 license, which wasn't harming the project in any way.

The open source license permits others to distribute the software for free and well packaged in the distro when he tries to sell it, so open source harms his (new?) business model.
The other issue is that open source does not care too much of the developer and some people sometimes don't like it (see [XFree86 license change with mandatory credits](https://en.wikipedia.org/wiki/XFree86#2004:_Licensing_controversy))

**Comment by dacap** [comment]
Please see that the "credit" issue is a minor issue, don't miss the main point: crash reports. If Linux distributions don't implement something like this, the whole license change would be completely useless. With this change I want to make a point: Linux users have benefits from Windows and OS X users, but the reverse isn't true. The main thing Linux distributions should solve is this: a program that crashes on Linux, should give so much information that I'd love to be on that Linux distribution.

**Comment by akien-mga** [comment]
@dacap Well, our experience differ, I codevelop a cross-platform application (Godot Engine), and the most helpful crash reports we get are clearly from Linux users, as many of them know how to use gdb to get a backtrace. And obviously, crashes fixed on Linux are often general crashes that affect all platforms, so Windows, OSX, Android, iOS, Emscripten, BlackBerry 10 and Co. benefit from it too.
It might have been a bother in your specific experience, but I find it a poor argument to generalise this way.

Your blog post has some other interesting arguments regarding how distros present their packages, but as I mentioned that hardly warrants going proprietary.

**Comment by JeromeMartinez** [comment]
> With this change I want to make a point: Linux users have benefits from Windows and OS X users, but the reverse isn't true. 

Seriously, no need to hide behind "mutual benefits" issue or something else like that: Linux does not harm you, you have just to answer "No support on Linux" if you want, so you don't spend time with such users. Actually, your license change has also an impact on Windows and Mac users (they can not distribute too), so that demonstrates that Linux users are not the problem.
The main issue looks like more in [your sentence](http://dev.aseprite.org/post/149797781837/new-source-code-license) "there is a feeling inside your head that makes you tremble: all your code is out there available for free" (weird, because it is the base of open source, and you chose an open source license 10 years ago).

Why not just saying something like "I don't like open source anymore, I prefer to sell per unit the old fashion way because I think it is the best business model for me now"?

**Comment by JeromeMartinez** [comment]
As keeping open source code is relatively important for me, I can offer:
- automatic builds (nightlies, releases) for Linux/Mac/Windows, digitally signed and so on, as I do for my own open source projects,
- host of the website,
- legal stuff,
In case you decide to keep Aseprite open source alive.

But as the main developer decided to retire from Aseprite open source development, it would be useless if there are no other active developers (Aseprite open source will just die a bit slower) and it looks like that there are currently no people thinking to actively develop Aseprite open source, so it would not worth to do it and it may be better to let Aseprite open source dying.

**Comment by dacap** [comment]
> the most helpful crash reports we get are clearly from Linux users, as many of them know how to use gdb to get a backtrace

Hi @akien-mga, it's clear to me that our end-users have a different skills.

> Why not just saying something like "I don't like open source anymore, I prefer to sell per unit the old fashion way because I think it is the best business model for me now"?

Hi @JeromeMartinez, we can ignore the whole post or we can see the issues in the Linux community. I could have wrote the blog post and keep Aseprite with a GPL license anyway, but will someone hear something? The answer is no, nobody cares what the developer have to say if the software is still GPL.

And please see the "So why not GPL?": *I don’t think anymore that distribution of software is a good thing just because it can be freely distributed without any cost. (Even from an ecological point of view.)*

**Comment by Sslaxx** [comment]
You may want to rename this fork to avoid potential issues in the future (trademarks and all that).

**Comment by Conan-Kudo** [comment]
@dacap For what it's worth, there are automatic crash reporting frameworks available out there. For example, the Red Hat/Fedora family has ABRT (Automatic Bug Reporting Tool), which applications can drop in configuration information to send crash reports to the right place. It also has a library (libreport) for doing this too.

I don't know what the state of ABRT packaging in most distributions is, but it is possible to use a tool like that to deliver crash reports automatically to the distro and the developer.

**Comment by alexander-b** [comment]
@dacap «I could have wrote the blog post and keep Aseprite with a GPL license anyway, but will someone hear something? The answer is no, nobody cares what the developer have to say if the software is still GPL.» — you're effectively arguing for a totalitarian regime. "People don't listen to me when I try to have a rational discourse, so I'm going to establish myself as a hierarchical power structure in an attempt at bullying them into submission." But that's not an effective tactic against people who believe that software must be free. Furthermore your assertion is utterly baseless anyway.

**Comment by dacap** [comment]
> you're effectively arguing for a totalitarian regime

We've a complete different view about what software is. Sorry.

> "[...] so I'm going to establish myself as a hierarchical power structure in an attempt at bullying them into submission."

😆  so I'm an insignificant human being from Argentina that are going to control all you people, sure.

I'll leave just this reference:
![vsptn3cpkahoa-1](https://cloud.githubusercontent.com/assets/39654/18254690/be9247ae-7376-11e6-90ec-5f8f899dc86c.gif)

**Comment by akien-mga** [comment]
I'd be glad if people would leave personal accusations or views on @dacap or what he did out of this topic/repo. @dacap is free to choose whatever he thinks best for himself and his software, and you guys should be glad that he did so much worked under the GPLv2 that we can now build on, instead of aiming for the Godwin point regarding a simple license change...

**Comment by dacap** [comment]
Anyway I'll leave an open question for you to think about (because I'm thinking about these issues in the last weeks):

1. free linux distributions have given a low entry point to run "virtual" machine/servers, and run software on them
1. the world doesn't have infinite resources
1. more servers running need more electrical power
1. more electrical power generates more CO2 emission

This is an actual problem I see, I don't know what the solution is, but the problem is there. "Infinite" free distribution of software looks like an ecological problem. I think we should start thinking about this issue now.

**Comment by alexander-b** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

On 05/09/16 19:48, Rémi Verschelde wrote:
> I'd be glad if people would leave personal accusations or views on
>  @dacap or what he did out of this topic/repo.
That doesn't make sense. He's the one who's brought up his baseless
assertion in this thread. If that is off-topic, then fine. But giving
him an arena to spout falsehoods without allowing replies does not seem
desirable, as I would assume he wants some feedback on his ideas.

> @dacap is free to choose whatever he thinks best for himself and
> his software
This also misses the point. You could say the same about any action.
Just because you think an action was good, that doesn't mean I have to
agree.

> and you guys should be glad that he did so much worked under the 
> GPLv2 that we can now build on, instead of aiming for the Godwin 
> point regarding a simple license change...
I am perfectly capable of both. He did lots of great work that we now
benefit from; that's great. But he also did something that I find
immoral, which is very unfortunate.
- -- 
Alexander
alexander@plaimi.net
https://secure.plaimi.net/~alexander
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iQIcBAEBCgAGBQJXzbIOAAoJENQqWdRUGk8Bc8EP/0dN+voJzy6tXI1cZ63ffoUT
6HX+OgycNjA2XD7stHXUh6K8Ri15PACT/JMTLXgMHNiZeknuBZ/pBJXw8kKjNXNF
jUD4vovdvAemB1uVxd3kQ6KO977b6lGNFoF44tDI+U6oaNL8fspZITuqXPErMgMQ
VwNv64mi/c7h+s5GRtNJ+HwTQel6OIdkoYAXI+t0DLFqmCpxX8Kpfhclf/x5fGks
AEzyX4cDeKBydHz6vNSoE5gSZFVBlZMOhD3WMJaQlnaIlcP1aLWiBO8iL7hMRwlw
eDzQBA8XGNALtjFjEGuybOQJJaDv4K2C7DAdQwKqTJU6dUPmCUbxa7EZ71rEoPjK
TK3+LgQ/VqFictZKtvGk/cSA5S11QeTZi9KKoJbxBgymNIjFTzmcfLRZ7EY8o4Vs
ikQa+RqdYCir93PqVLVkAtV0daX8Pct9TDBivuiVA8gaUnKFX5XgNQGvcLfATCgs
Qjtr3eCrZ15H95VWyxZx2CQdB8Ep3pZ++FyFd9OwV1QsuBWsanPYlaeX46SSLETf
66ipnfXq3H+cSej/YTejd+zYJRTF8BMBMn23GlxWzYR3R18AdA7WHAHDswFUJz6D
QFaeweLzuQTn1AMEUOaiZI6FZvvA6+MhqtTBdaBPznZ/DvAm3cqZ+U3edNrMkjQZ
904c2wHfKlhBEHd2DmwL
=be2r
-----END PGP SIGNATURE-----

**Comment by JeromeMartinez** [comment]
@dacap you don't like open source anymore, that's your choice.
But you don't need to invent excuses for not liking it.
There is absolutely no relationship between open source and CO2 emission.
This is insulting (actually, you already insulted a lot open source with crazy excuses for leaving it)

Looks like you are trying to find so many excuses against open source that we can only think that you know yourself that your intentions are not so noble (and surprise! you are the only one who can make money from the code now...)

**Comment by alexander-b** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

On 05/09/16 19:52, David Capello wrote:
> 1. free linux distributions have given a low entry point to run 
> "virtual" machine/servers, and run software on them
> 
> 1. the world doesn't have infinite resources
> 
> 1. more servers running need more electrical power
> 
> 1. more electrical power generates more CO2 emission
> 
> This is an actual problem I see, I don't know what the solution is,
>  but the problem is there. "Infinite" free distribution of software
>  looks like an ecological problem. I think we should start thinking
>  about this issue now.
I agree completely. This is exceedingly important. But the solution
isn't to take away user freedom. And software freedom isn't even the
problem in the first place.
- -- 
Alexander
alexander@plaimi.net
https://secure.plaimi.net/~alexander
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iQIcBAEBCgAGBQJXzbKXAAoJENQqWdRUGk8BkpUP/1/8Ktdqo813wiJtTWhdgeBF
sAoVuQYR3fqsJrnqNxQ+AJ/YBh5cDUAyYenZXwLOGpqmInAT4SQaQbKwMwMlRDGO
jO0a7n6LkC2E+vA5VWDcSPhwad8y6G9MRpJfPJO1rWRZYXJ/N2ELNYwdRLN9TARV
GdCJOOaaBY+t9gS461trgtTfM5PoB8yLuCNGPwLJnVQw1mnVRAWq3o/Ce39woHsC
BWOeS+iyYx0FltZkE+mGhMz6LXD8ZbdeqNP5aKxWwYiq7gBF2uEahjxTaAdoppAX
dMnVxf2yZqknphNetXfzIr3bVoUt1fV9X5Ie/mD8ExlnmqT5PTANO1KoaSjsFNoK
j3BxxtRDgW42VP8vH9wHemCoyqRM/W8t3wuEBTKWkDn+bY2YexTWyfD2QV/ODh3U
6wK4kprOgr9pbkLsgN37vpFVvRyAXql2HYPE35P9V8CZg2b1eDBGbRzvRjFQacDS
5tZ1WAGei9ndtvxG6InhoCIEaAUMdfyV2oEgzl6cT2r8P+2keF80c6/46biVdSyB
SO5nT29q02anpbPo3hm+MRNh6bRmZ/ipxahaF5PD5j+S4A/2M3jwXHnaCNQ4phrX
6VV50Qq0sirt4lT+KHdsy4jEagHqoXhg8qkLHxcarGaB+AgWzyxO8vBMfvYHTX+C
GWl6HkIkl0CxyFT8C6EP
=6wi3
-----END PGP SIGNATURE-----

**Comment by alexander-b** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

On 05/09/16 19:58, Jérôme Martinez wrote:
> and surprise! you are the only one who can make money from the
> code now...
Only whatever happens in aseprite from
5ecc356a41c8e29977f8608d8826489d24f5fa6c and forward. Remember, all
the rest is still free software. The GPL empowered us by making this
fork possible, and David gave us all this wonderful free code that we
can use.
- -- 
Alexander
alexander@plaimi.net
https://secure.plaimi.net/~alexander
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iQIcBAEBCgAGBQJXzbNEAAoJENQqWdRUGk8BzrgQAMdMB+uvUxqCm7p7e9sRZ+wN
wm9gg4s1XGK0w5DYhKkk7i8caJMMxya03oYQIsiZhY6RcDnQNU/o9nFDI0B3OTAc
GHIrkJ5oAUl8FqOaWqwEnnOF7q1Z/tPULL9znZW9fnu+eZBPjp3Dge2DZHLf7bb9
AnIEMgo00QM8npYFPgHvUtdoCDb9O6vXoKFKhcoLj8CEJCIpaqrTK2bGU+aRNqls
xa/XDei9trciOv3m38WWZbTDXLXNvhygx5dYww5xhLcdcGDuWysG4Frc/JfCNe5i
k1tLEyV9UmAD6zwpKl7sHkvtTyS+Mt6PYIb52+Z2IY/nvUOQpVZOJCxodzP7rveo
wT8maSu5ZFa0T/fqRo+V0jFKmvyF9F3XLY5CjT8a7Ndj6EhPUqjNRaCV5vorizxw
/5Ux8c4GgfUy6sjhMNDQwFfAW9z/8eiUfjZd+/qIk/DdHfJ3ItmfijJ06ORGZhUA
a/PUuzVpR9VLZc9h5Qu3mAKr+K5ketS50KS87ZwqAfmXTz+TVSwM2Evu98FFphF+
0iambfdfeFY0iDxRRgcCiEcn5Ar7cmWRYqV3BsQ4tP67LCoVtuAh2PNeOaM4Iqpa
uUjo7upXt6LkF2A0ekkIXCfmRmwbQXo1Ky8ksDsEfhTwzwdAoIyKkWP6ypWA7IRs
I/n/aBYrWg68ciUsp8Zl
=YoM+
-----END PGP SIGNATURE-----

**Comment by JeromeMartinez** [comment]
@alexander-b 

> Only whatever happens in aseprite from 5ecc356 and forward.

Obviously, I was speaking of future developments not under GPL.
Thanks to the so hated open source, other people are able to fork and keep the open source version alive, this is the argument I say to my users ("if you don't think I am the right person anymore, you can fire, the code belong to users and not the the developer, users have vote right").

(but reminder: for the moment, the fork has no developer, so it will just slowly die)

**Comment by dacap** [comment]
> @dacap you don't like open source anymore, that's your choice.

I'm not saying this. I like open source, I like free software. There are great ideas behind these movements. Aseprite will have still its source code "open" (from an alternative definition of what "open" is). The main issue I'm talking about here and in my blog post is this: "redistribution of software has a cost," it's not so "free" as we might think it is. And we can take "cost" from any point of view: price (execution and hardware construction), bugs propagation, bad user experience, lost of art/personal creations, natural resource usage, responsibility of damages, etc.

> Looks like you are trying to find so many excuses against open source that we can only think that you know yourself that your intentions are not so noble

What would not be a "not so noble" intention for you? What I'm seeing is that what you call "excuses" are in fact real issues that you don't want to look at (or are trying to avoid). But that's fine, you can fight for your own causes, and I can fight for my causes. We don't have to destroy each other, right? Also, I want to be clear here: I don't have the solutions to the problems I'm describing.

> But the solution isn't to take away user freedom.

I cannot ensure you at this moment if "user freedom" (or even "developer freedom") is the most important thing that matters on software. Complete freedom might looks great, but what if that "freedom" means "massive destruction."

> And software freedom isn't even the problem in the first place.

Yes, commercial software has the same issue. But free software encourage free distribution without any kind of reasoning about the cost of running software.

**Comment by JeromeMartinez** [comment]
@dacap 

> "redistribution of software has a cost," it's not so "free" as we might think it is.

Seriously? my own software is redistributed thousands times per day, and I don't see what you are talking about.
Please don't find again excuses: open source does not force you to answer to bug / support requests.
If you think about such cost, easy: keep it open source and don't answer to bug / support requests.
If you think bandwidth: distros do it for you, $0 for you.
where is the cost?

So please stop to accuse open source about what it is not responsible.
And if you really like open source, keep your code open source. Else that means that your words are in sync with facts.

Sorry but as I already said, you are looking for excuses which are not about open source. What you wrote on your blog post is insulting open source because you accuse it of something it is not responsible. Please stop. Leaving open source is your choice, accusing it of what it is not responsible is finding excuses.

> Complete freedom might looks great, but what if that "freedom" means "massive destruction."

Insulting open source, again. Weird that you say that you like it.

> But free software encourage free distribution without any kind of reasoning about the cost of running software.

Because this is not the purpose of open source.
(and running software has 0 cost for you. Reminder: open source does not force you to provide an email address, if you like open source and want to avoid complains from people using it, just remove your email address, not the GPL)

> What would not be a "not so noble" intention for you? 

Remind that I say "not so noble" from your point of view, as you are the one who tries to hide behind false excuse.
intention is pretty clear: $, as much as you can.
Note that this is not "not noble" from my point of view, I make money from my (open source) software. I just don't hide behind excuses when I say "if you want this bug fixed, pay me".

***

Looks like we can not really discuss, because you hide behind excuses every time we demonstrate that your words are not in sync with your actions (and actually your words are also not in sync: you say that you like open source which means that you like free distribution, then you say you dislike free distribution).

Really, read again your sentences: you like open source and you think that open source is "massive destruction" (later, we'll learn that open source kills cats). How can we debate with that?

Think again about your goals, then decide about the right actions. If open source does not fit, fine, say it and stop to say that you like open source. And the most important: don't accuse open source about unrelated things.

***

For the moment, removing your email address is enough for fixing your criticisms on your blog (for the new "CO2" stuff, this is so crazy that we can not debate), no need to change the license. But I bet that $ with the control of the distribution are more important (if not, really, just remove your email address and/or answer "support only for users who paid and use my versions").

**Comment by dacap** [comment]
@JeromeMartinez I see that you are confusing "open source" with "free software" with "user freedom to redistribute." Finally, if you think a software is "redistributed thousands times per day" without any cost (and I'm not talking about money!), you are just not living in planet Earth.

**Comment by JeromeMartinez** [comment]
@dacap sorry, just wanting to point out a crazy thing:

> And we can take "cost" from any point of view: price (execution and hardware construction), bugs propagation, bad user experience, lost of art/personal creations, natural resource usage, responsibility of damages, etc.

The best method against that is to avoid to write code and avoid to sell your product (each sell is more CO2!)
From your arguments, if you believe in it, the only thing you can do is to stop to code, stop to sell your product at all. Will you do it?
Surprisingly, you don't do that.
Sorry, but I can not believe the reasons you give because your actions would be completely different if they are the real reasons.

I also note that when we "kill" the arguments you first provided, you find new ones, like if you find them only now.

And now:

> I see that you are confusing "open source" with "free software" with "user freedom to redistribute."

Please stop to say you like open source. user freedom to redistribute is one of the 4 liberties of open source. this is your right to dislike open source, but saying that you like open source but you dislike user freedom to redistribute is a lie (also to yourself if you really believe what you say).

***

You say that you remove open source license for creating a debate, but that convince nobody because the first thing you do is saying that you hate open source (yes, if you hate "user freedom to redistribute.", you hate also open source because open source includes that).

***

Summary: we will not be able to convince you to go back to open source because the issue is not open the excuses your provide, and open source is definitely not compatible with your (new) intentions about what you want to do with your software.
This is your right, and you will probably be able to do it (because there is no developer interested in continuing on the open source version so it will not compete), but that is a pity that you hide behind false excuses for doing that, accusing something which never hurt you.

Please do what you want about the proprietary version of your software (which will kill trees due to CO2 you consume for distributing it, and also the CO2 used for building your development computer), but stop to hide behind false excuses, discussion would be more honest.

**Comment by dacap** [comment]
> I also note that when we "kill" the arguments you first provided, you find new ones, like if you find them only now.

What?! When? Where?

> > I see that you are confusing "open source" with "free software" with "user freedom to redistribute."
>
> Please stop to say you like open source. user freedom to redistribute is one of the 4 liberties of open source. this is your right to dislike open source, but saying that you like open source but you dislike user freedom to redistribute is a lie (also to yourself if you really believe what you say).

About the 4 liberties, you are talking about [the 4 liberties of free software as defined by the FSF](https://www.gnu.org/philosophy/free-sw.en.html) (which is completely different to what "open source" is). About "open source" we should refer the definition given by [The Open Source Initiative](https://opensource.org/osd). Those are "definitions." I don't see why I cannot start a debate in one point of those definitions (redistribution) if I'm not 100% agree with it.

**Comment by JeromeMartinez** [comment]
> I don't see why I cannot start a debate in one point of those definitions (redistribution) if I don't agree 100% with it.

OK, you want to say "I like x with x being my definition which conflict at 99% with the commonly accepted definition of x". It becomes so funny.
Looks like you want to avoid the debate about open source, and looks like this is not the purpose of the change ($, $, $... Which looks fine for you, lot of your users don't care about open source so you don't lose too much).

the remaining question is: will an open source version survive the retirement of the main developer from open source version? For the moment no developer is really interested in it publicly so...

**Comment by Conan-Kudo** [comment]
@dacap For what it's worth, when Bruce Perens drafted the Open Source Definition, he explicitly included the Free Software's Four Freedoms in it. In fact, the Open Source Definition is designed to be a clarification of the Four Freedoms. They are supposed to be effectively synonymous.

**Comment by dacap** [comment]
@Conan-Kudo thanks for the clarification. I'll quote [this article](https://www.gnu.org/philosophy/open-source-misses-the-point.html) from Stallman, where he says _"The official definition of 'open source software' was derived indirectly from our criteria for free software. It is not the same; it is a little looser in some respects. Nonetheless, their definition agrees with our definition in most cases."_ so let's say that I've exaggerated a lot saying *"which is completely different to what "open source" is."*

**Comment by alexander-b** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

On 05/09/16 20:33, David Capello wrote:
> I cannot ensure you at this moment if "user freedom" (or even 
> "developer freedom") is the most important thing that matters on 
> software. Complete freedom might looks great, but what if that 
> "freedom" means "massive destruction."
I already identified your thinking as totalitarian. I think that is
still accurate. It sounds to me like you don't want people to have
freedom if that freedom leads to bad things, and would prefer some
powerful entity to dictate their matters instead. I think you'll
struggle to find historic evidence where this has worked out well.

> Yes, commercial software has the same issue. But free software 
> encourage free distribution without any kind of reasoning about the
> cost of running software.
So does proprietary software. And I am willing to suggest that Google,
Facebook, and other major multinational corporations, are likely to
outdo all free software put together, with regards to environmental
issues.

- -- 
Alexander
alexander@plaimi.net
https://secure.plaimi.net/~alexander
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iQIcBAEBCgAGBQJXzpVjAAoJENQqWdRUGk8BqVkQAIRnKJCC6QWet0DIIMTdTvxU
jO3gpb5t5nS02Ds7sZdvJSeWsFhWwlFFK7EqWndjlRATAG3DeDEew2yIKLxm+ajG
SctW91mmE5opEzISt8CmXO79uoHWvtRzgwTwo3170ZxvoWL8UUEw5BuUWyGwwN0R
+DCe0+muNtCUlZDJYJcc8sS8L00QGBWheiTEVVcTYsRRdggsjGFVWsHsHTf9dt1Y
3v4SOpa6T+ySnDcTFLqOGjN2iLCE33LDB9Z+LYe5DTVue4F4zCZiP/HvzNwMJ8LJ
NOwwedzchVA/JXMfWqSdEQHDL6kxeKTWy0vm/NcZBZQvlZegQp/53mqv+zTqFx4B
ZkTYm8NxIYJG62L3FdcIAqIxyedGE7Ctkw3bpNHZmTLVxVCbdzbH+QV/dhU89q5i
nBLvr2LlVblGjQuZBQWi79085pS4SRIe2zguSXJZLEekPjCb9dweu217AKIMEMGF
ehBfxD6RN5FczLc+KN/m4zej7QrrNca5F592sp8gtzYXf83o6SWuG34l9IXjgMtQ
1aTspZtwYKng9w2LXyR9ebIbu6lqWlrj8pvatTJVCa+aO4JUf2riUrWjFLs6OT7D
9UsYc9IhbJHhnSHOrO4fn47O3Y8DZ8Y2G2a5oY6cIzB32vldaLn83rjGSeH3Esk4
TDiT/KLgGYoD4fQ05fFg
=TAcy
-----END PGP SIGNATURE-----

**Comment by akien-mga** [comment]
I'll ignore the above flamewar as it's fully outside the scope of this issue.

Some time as passed since the fork, and some potential contributors have expressed interested in a community-maintained continuation fork of the GPLv2 code base.

As it stands currently, I don't have time myself and don't see many hands raised to really develop this project actively and make it evolve to be something new, distinctive from @dacap's original project. I would therefore propose to keep our objectives humble, and to make this fork a maintenance repo for the code as it was before the fork, i.e.:
- we fight bitrot and ensure that the code builds in the future
- we improve things here and there to ease packaging (which would bring us additional distro contributors that could help unbundle libraries if need be, etc.)
- we fix bugs that get reported to us if we know how to
- we do some improvements when we feel like it, but most likely without the vision and dedication of the original project

The result would be that this repo would gradually evolve into a slightly better open source project, though it wouldn't (at least under my direction, but I'd gladly hand over the repo to someone with more time and interest) evolve to become a fully different program. The objective would be that at least Linux distros replace their aseprite package with this fork, so that they can continue providing this nice software with a maintained open source version.

Of course, this well-maintained repo could then be used as a basis for some bigger development efforts if there is interest, provided some contributors are up to the task and willing to lead such efforts. So this "limitation" of the objectives that I propose is of course not definitive, but just my short-term aim.

The next steps to really get started would then be to fully rebrand the project (see #8 for naming proposals), and maybe also change its versioning scheme to avoid conflicting with that of @dacap's project.

**Comment by jjhaggar** [comment]
Hello, is this fork still alive? I hope it is, because although I support the original Aseprite by @dacap (I bought it and I recommend it to all my friends), I think a Libre version would be of benefit for everybody :)
Any plan on providing binaries?

**Comment by Symbian9** [comment]
> I think a Libre version

@jjhaggar, Yeah, this is libre version/fork that should be renamed to *LibreSprite*

- https://github.com/aseprite-gpl/aseprite/issues/8

Development not so active as original proprietary *Aseprite*, but project now in re-branding stage and all is WIP

- https://github.com/aseprite-gpl/aseprite/commits

Pull requests are welcome! ;-)

- https://github.com/aseprite-gpl/aseprite/pulls

**Comment by Wuzzy2** [comment]
Hi there, here's another old Aseprite user here.
First of all, **THANK YOU** for preserving this piece of software and forking it very early. This was an important step to make.

I think the top goal should be first to simply *make it work* for users. I do not think just keeping it a read-only repository is a good idea. Anyone can do that already by simply grabbing the relevant commit, so this is not really useful. Being an *usable* software should be top priority.

Then rebranding is also important. The confusion is just too high otherwise.
The project origin should be properly documented (but not advertised), maybe even with the commit ID that started at all (important for developers). I think a few sentences would be enough. Credit to all original authors is mandatory for obvious reasons.
Then I would remove features and widgets which only really make sense in original Aseprite, but not in this fork. The Aseprite news are a perfect example, they have no place in this fork and should be removed. URLs, about dialog and other references need to be adjusted as well.
The program should appear to stand on its own, but its historic ties should not be denied (without advertising them).

If this software has been rebranded and is reasonably stable, **spread the software**. Alert all package maintainers of GNU/Linux distrubutions (**very** important), spread awareness, talk to people about it, etc. I think many maintainers would be more than happy to include a free fork of Aseprite. The first distros to alert would be “hardcore” free software distros like Parabola, gNewSense,  etc.

So people need to know that there's a free software fork.

I agree a small website would be nice, to make it look like a serious project. To safe time, you could start with a very simple static website with a few links and screenshots, even this could help a ton.

And after that, the future of this project is pretty open, I guess. But I think in the long run the goal should be to make an usable software which is also free. I don't think adding new features should be rejected *on principle*.

Finally, I strongly recommend to not work on fancy features before the rebranding work has been done.

**Comment by aggsol** [comment]
Also [slate](https://github.com/mitchcurtis/slate/) is a great FOSS project that could also benefit from contribution.

**Comment by RUSshy** [comment]
This repo shouldn't exist, shame on you

**Comment by RUSshy** [comment]
You are all a bunch of thieves

**Comment by JamesK89** [comment]
Not according to the license. This is an open source project forked from _the last OPEN SOURCE version_ Asesprite. In spirit of open source skilled developers wish to keep Asesprite open under the original GNU GPLv2 license without the burden of a proprietary license. You're entitled to your opinion but, please, be constructive if you do have anything to say.

**Comment by JeromeMartinez** [comment]
> This repo shouldn't exist, shame on you

Why? All is compatible with the author wish (he gave people the right to fork).

> You are all a bunch of thieves

I guess you misunderstood something: this is 100% legal, thanks to the rights the author of the software granted to people who receive the software (he does not provide new code with same rights, but this is not an issue for a fork as old code can not have license removed, as stipulated in the license the author gave to people).

Long story short: you do a mistake when you think there are thieves here, as all is legal.

**Comment by Symbian9** [comment]
Few latest discussions on HN and Reddit:

- https://news.ycombinator.com/item?id=17343864
- https://old.reddit.com/r/linux/comments/8s8hfj/libresprite_fork_of_the_last_gplv2_commit_to/

**Comment by blurymind** [comment]
Open source software allows being integrated in more interesting ways and development is more democratic- as the users can directly add features to it- if they so have the know how.

I recently bundled piskel inside gdevelop's new ide- via electron. The license of both applications and underlying technology allowed me to do that. But in the process I also got help from both authors of gdevelop and piskel. This collaboration of me- the user, and the two more experienced developers allowed a very seamless integration of piskel into gdevelop. Now it is a part of gdevelop and its great as the two compliment each other very well. 

That would have never happened without the license being open source. It allowed me to study how both are built, learn from their authors- and in return- contribute new features. That made the experience of contributing to it wonderful.

In that spirit- I believe that prior to losing its open license- aseprite must have had collaborators that contribute to it because they love being a part of its open source development. They still have this fork and should be allowed to continue to built on top of it- and yes even compete. Even if the author does not necessary wish to follow godot's model of funding- via patreon- that doesnt mean that the fork should die.
I think that if there is enough desire, this fork can evolve into something wonderful or even be apart of another open source project :) 
It makes me to see it here on github and hope for the best

**Comment by RUSshy** [comment]
@GlaDOSik they are forks because i contributed to them, you are ignorant

this fork steal years of work and they changed the name without even puting a contribution on them

they are just stealers and on top of that attention whores

**Comment by GlaDOSik** [comment]
Yes, that was just a quick ignorant observation. But what is the difference? Still, contributor or not, you have 21 forked projects - years and years of work of many people - linked to your account. So why should this be a stealing? Because of fact the new code is no longer GPLv2? You cannot prevent others to re-distribute a code which was released under permissive license. And from what I can tell, contributions are all over the place. I'm ignorant and you're acting immature.

**Comment by akien-mga** [comment]
Let's cool down a bit... I'll unlock the issue in a few days and see how it goes.

**Comment by RealMizat** [comment]
If I simply had more experience I would love to work on this.

**Comment by RealMizat** [comment]
If I simply had more experience I would love to work on this.

**Comment by jjhaggar** [comment]
@akien-mga It seems that a new maintainer has appeared! :D
https://github.com/LibreSprite/LibreSprite/issues/40

**Comment by maujin111** [comment]
If you need help translating it to Spanish i would love to offer my self for the job, cloud be the ṕrogram or the website, readme, whatever
